src/main/java/org/italiangrid/storm/webdav/server/servlet/StoRMServlet.java

```diff
@@ -106,6 +107,23 @@ protected String getEncodedPathInContext(HttpServletRequest request, boolean inc
   protected void doGet(HttpServletRequest request, HttpServletResponse response)
       throws ServletException, IOException {
     SciTag scitag = (SciTag) request.getAttribute(SciTag.SCITAG_ATTRIBUTE);
+    if (serviceConfig.getNginxReverseProxy()) {
+      final boolean included = request.getAttribute(RequestDispatcher.INCLUDE_REQUEST_URI) != null;
+      final String pathInContext = getEncodedPathInContext(request, included);
+      String resolvedPath = pathResolver.resolvePath(pathInContext);
+      File f = new File(resolvedPath);
+      if (f.isFile()) {
+        if (scitag != null) {
+          SciTagTransfer scitagTransfer = new SciTagTransfer(scitag, request.getServerName(),
+              request.getServerPort(), request.getRemoteAddr(), request.getRemotePort());
+          scitagTransfer.writeStart();
+          response.setHeader("X-SciTag-actId", Integer.toString(scitag.activityId()));
+          response.setHeader("X-SciTag-expId", Integer.toString(scitag.experimentId()));
+        }
+        response.setHeader("X-Accel-Redirect", "/internal-get" + resolvedPath);
```

Comment:
I would add a constant with the internal prefix `/internal-get`, with a comment that this should match the location in the nginx configuration.